src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -4418,19 +4442,114 @@ def generate_agg_qc(
                 )
 
             else:
-                for agg_func, cols in agg_func_to_col_map.items():
-                    col_single_agg_func_map = {
-                        column: AggFuncInfo(
-                            func=agg_func if column in cols else "min",
-                            is_dummy_agg=column not in cols,
+                if uses_named_aggs:
+                    # If this is true, then we are dealing with agg with NamedAggregations.
+
+                    def generate_single_agg_column_func_map(
+                        identifier_pair: PandasLabelToSnowflakeIdentifierPair,
+                        agg_func: AggFuncWithLabel,
+                    ) -> dict[PandasLabelToSnowflakeIdentifierPair, AggFuncInfo]:
+                        """
+                        Helper function to produce the agg func map for a single aggregation, including dummy aggregations.
+
+                        Notes:
+                        Adds dummy aggregations for all columns that will be aggregated over.
+                        """
+                        col_single_agg_func_map = {}
+                        for c in column_to_agg_func.keys():
+                            if c == identifier_pair:
+                                col_single_agg_func_map[c] = AggFuncInfo(
+                                    func=agg_func.func, is_dummy_agg=False
+                                )
+                            else:
+                                col_single_agg_func_map[c] = AggFuncInfo(
+                                    func="min", is_dummy_agg=True
+                                )
+                        return col_single_agg_func_map
+
+                    # We can't simply append the generated_qcs to single_agg_func_query_compilers, because
+                    # we have changed the order of the aggregations - see the comment in the function
+                    # extract_validate_and_try_convert_named_aggs_from_kwargs for an explanation of how
+                    # and why the order changes. Instead, we get a mapping of the name of the aggregation to the
+                    # QueryCompiler it produces, which we can then use to insert into single_agg_func_query_compilers
+                    # in the correct order, which we recieve from the frontend, so that when the concatenation occurs,
+                    # the final QueryCompiler is ordered correctly.
+                    index_label_to_generated_qcs: dict[
+                        Hashable, list["SnowflakeQueryCompiler"]
+                    ] = {}
+                    for (
+                        identifier_pair,
+                        agg_func_with_label,
+                    ) in column_to_agg_func.items():
+                        if isinstance(agg_func_with_label, list):
```

Comment:
Can you convert agg_func_with_label to a one-element list if it's a scalar to avoid the duplicated code here?